code/modules/lighting/lighting_source.dm

```diff
@@ -205,49 +216,82 @@
+	for(var/x in (-(bound_range) + x_offset - 0.5) to (bound_range + x_offset + 0.5))
 		var/list/row = list()
-		for(var/y in (-bound_range - 0.5) to (bound_range + 0.5))
-			row += falloff_at_coord(x, y, z_level, range)
+		for(var/y in (-(bound_range) + y_offset - 0.5) to (bound_range + y_offset + 0.5))
+			row += falloff_at_coord(x, y, z_level, range, center_dir, light_angle)
 		encode += list(row)
 	return encode
 
 /// Returns a THREE dimensional list of lists that encodes the lighting falloff of our source
 /// Takes anything that impacts our generation as input
 /// This function should be "pure", no side effects or reads from the passed object
-/datum/light_source/proc/generate_sheet_multiz(range)
+/datum/light_source/proc/generate_sheet_multiz(range, visual_offset, x_offset, y_offset, center_dir, angle)
 	var/list/encode = list()
 	var/z_range = SSmapping.max_plane_offset // Let's just be safe yeah?
 	for(var/z in -z_range to z_range)
-		var/list/sheet = generate_sheet(range, z)
+		var/list/sheet = generate_sheet(range, visual_offset, x_offset, y_offset, center_dir, angle, z)
 		encode += list(sheet)
 	return encode
 
 /// Takes x y and z offsets from the source as input, alongside our source's range
 /// Returns a value between 0 and 1, 0 being dark on that tile, 1 being fully lit
-/datum/light_source/proc/falloff_at_coord(x, y, z, range)
-	var/_range_divisor = max(1, range)
+/datum/light_source/proc/falloff_at_coord(x, y, z, range, center_dir, angle)
+	var/range_divisor = max(1, range)
+
 	// You may notice we use squares here even though there are three components
 	// Because z diffs are so functionally small, cubes and cube roots are too aggressive
-	return 1 - CLAMP01(sqrt(x ** 2 + y ** 2 + z ** 2 + LIGHTING_HEIGHT) / _range_divisor)
+	// The larger the distance is, the less bright our light will be
+	var/multiplier = 1 - CLAMP01(sqrt(x ** 2 + y ** 2 + z ** 2 + LIGHTING_HEIGHT) / range_divisor)
+	if(angle >= 360 || angle <= 0)
+		return multiplier
+
+	// Turn our positional offset into an angle
+	var/coord_angle = delta_to_angle(x, y)
+	// Get the difference between the angle we want, and the angle we have
+	var/center_angle = dir2angle(center_dir)
+	var/angle_delta = abs(center_angle - coord_angle)
+	// Now we have to normalize the angle delta to be between 0 and 180, instead of 0 and 360
+	// This ensures removing say, 15 degrees removes it from both sides, rather then just one
+	// Turns an unfurling fan into a pair of scissors
+	if(angle_delta > 180)
+		angle_delta = 180 - (angle_delta - 180)
+	// We allow angle deltas to a certian amount, angle / 2
+	// If we pass that, then it starts effecting the visuals
+	// Oh and we'll scale it so 30 degrees is the "0" point, where things become fully dark
+	// This could be variable, it just isn't yet yaknow?
+	return max(multiplier * (1 - max(angle_delta - (angle / 2), 0) / 30), 0)
+
+/datum/light_source/proc/print_sheet()
```

Comment:
what is a sheet?

Reply:
I'll doc it but uh
Sheets are 2 dimensional lists that encode all the possible lighting values for a set of light vars (angle, dir, intensity, range)
We access them to pull out multipliers when passing out light values to corners.